Userland/Shell/Builtin.cpp

```diff
@@ -1049,16 +1049,51 @@ ErrorOr<int> Shell::builtin_time(Main::Arguments arguments)
 ErrorOr<int> Shell::builtin_umask(Main::Arguments arguments)
 {
     StringView mask_text;
+    bool symbolic_output = false;
 
     Core::ArgsParser parser;
+
+    parser.add_option(symbolic_output, "Produce symbolic output", "symbolic", 'S');
     parser.add_positional_argument(mask_text, "New mask (omit to get current mask)", "octal-mask", Core::ArgsParser::Required::No);
 
     if (!parser.parse(arguments, Core::ArgsParser::FailureBehavior::PrintUsage))
         return 1;
 
+    auto parse_symbolic_digit = [](int digit) -> ErrorOr<String> {
+        StringBuilder builder;
+
+        if ((digit & 4) == 0)
+            builder.append('r');
+        if ((digit & 2) == 0)
+            builder.append('w');
+        if ((digit & 1) == 0)
+            builder.append('x');
+        if (builder.is_empty())
+            builder.append('-');
```

Comment:
Nit: You can use `try_append()` here since this returns `ErrorOr<String>` anyway